src/watts/plugin_moose.py

```diff
@@ -205,3 +211,24 @@ def postrun(self, params: Parameters) -> ResultsMOOSE:
         inputs = ['MOOSE.i']
         outputs = [p for p in Path.cwd().iterdir() if p.name not in inputs]
         return ResultsMOOSE(params, time, inputs, outputs)
+
+
+    def _convert_unit(self, params: Parameters):
+        # Function to convert units to SI
+        u.imperial.enable()
+        self.params_copy = copy.deepcopy(params)
+
+        for key in self.params_copy.keys():
+            if key in self.params_copy.convert_param:
+
+                temperature_units = ['Kelvin', 'Celsius', 'Rankine', 'Fahrenheit',
+                                    'deg_C', 'deg_R', 'deg_F']
+
+                # Unit conversion for temperature needs to be done separately because 
+                # astropy uses a different method to convert temperature.
+                # Variables are converted to SI by default.
+
+                if self.params_copy[key].unit in temperature_units:
+                    self.params_copy[key] = self.params_copy[key].to(u.K, equivalencies=u.temperature()).value
+                else:
+                    self.params_copy[key] = self.params_copy[key].si.value
```

Comment:
I agree with @nstauff that we can push this into the general `Plugin` class. To have plugin-specific unit systems, this could be changed to something like:
```Python
if self.params_copy[key].unit in temperature_units:
    self.params_copy[key] = self.params_copy[key].to(self.unit_temperature, equivalences=u.temperature()).value
else:
    self.params_copy[key] = getattr(self.params_copy[key], self.unit_system).value
```

and then in the plugins themselves:
```Python
class PluginMOOSE(TemplatePlugin)
    """..."""

    unit_system = 'si'
    unit_temperature = 'K'

    def __init__(self, ...):
```
and
```Python
class PluginOpenMC(Plugin):
    """..."""

    unit_system = 'cgs'
    unit_temperature = 'K'

    def __init__(self, ...):
```